database/redis.go

```diff
@@ -0,0 +1,74 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package database
+
+import (
+	"context"
+	"strings"
+
+	"github.com/ava-labs/avalanchego/utils/logging"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/redis/go-redis/v9"
+	"go.uber.org/zap"
+)
+
+var _ RelayerDatabase = &RedisDatabase{}
+
+type RedisDatabase struct {
+	logger logging.Logger
+	client *redis.Client
+}
+
+func NewRedisDatabase(logger logging.Logger, redisURL string, relayerIDs []RelayerID) (*RedisDatabase, error) {
+	opts, err := redis.ParseURL(redisURL)
+	if err != nil {
+		logger.Error("Failed to parse Redis URL", zap.Error(err), zap.String("url", redisURL))
+		return nil, err
+	}
+
+	// Create a new Redis client.
+	// The server address, password, db index, and protocol version are extracted from the URL
+	// Request timeouts use the default value of 3 seconds
```

Comment:
Looking over `PraseURL` documentation, I think this can be set in the redis url as well, and not always defaulting to 3 seconds. `redis://user:password@localhost:6789/3?dial_timeout=3&db=1&read_timeout=6s&max_retries=2`

Reply:
If the timeout query parameters are not present, then the `opts` will be use the default value of 3 seconds. If the user passed in the URL above, then `opts` would use those timeouts instead.

Reply:
nit: I'd suggest adding that clarification in the comment. The comment sounds as if it always uses 3 seconds regardless of url

Reply:
Clarified